exp/controllers/azuremachinepool_controller.go

```diff
@@ -300,14 +300,15 @@ func (ampr *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, mac
 	err := machinePoolScope.InitMachinePoolCache(ctx)
 	if err != nil {
 		if errors.As(err, &reconcileError) && reconcileError.IsTerminal() {
-			ampr.Recorder.Eventf(machinePoolScope.AzureMachinePool, corev1.EventTypeWarning, "SKUNotFound", errors.Wrap(err, "failed to initialize machine cache").Error())
-			log.Error(err, "Failed to initialize machine cache")
+			ampr.Recorder.Eventf(machinePoolScope.AzureMachinePool, corev1.EventTypeWarning, "SKUNotFound", errors.Wrap(err, "failed to initialize machinepool cache").Error())
+			log.Error(err, "Failed to initialize machinepool cache")
+			// TODO: Should this be InvalidConfigurationMachinePoolError instead?
```

Comment:
Yes

Reply:
I think we would need to change the type of `AzureMachinePool.Status.FailureReason` in order to fix this. I can remove the TODO and open a follow up PR if that works for you.

Reply:
SGTM